src/pybullet_helpers/geometry.py

```diff
@@ -0,0 +1,75 @@
+"""PyBullet helper class for geometry utilities."""
+from __future__ import annotations
+
+from functools import cached_property
+from typing import NamedTuple, Tuple
+
+import numpy as np
+import pybullet as p
+from pybullet_utils.transformations import euler_from_quaternion, \
+    quaternion_from_euler
+
+from predicators.src.structs import Array
+
+Pose3D = Tuple[float, float, float]
+Quaternion = Tuple[float, float, float, float]
+RollPitchYaw = Tuple[float, float, float]
+
+
+class Pose(NamedTuple):
+    """Pose which is a position (translation) and rotation.
+
+    We use a NamedTuple as it supports retrieving by integer indexing
+    and most closely follows the PyBullet API.
+    """
+    # Cartesian (x, y, z) position
+    position: Pose3D
+    # Quaternion in (x, y, z, w) representation
+    orientation: Quaternion = (0.0, 0.0, 0.0, 1.0)
+
+    @classmethod
+    def from_rpy(cls, translation: Pose3D, rpy: RollPitchYaw) -> Pose:
+        """Create a Pose from translation and Euler roll-pitch-yaw angles."""
+        return cls(translation, quaternion_from_euler(*rpy))
+
+    @cached_property
```

Comment:
it turns out that `cached_property` actually throws an error here, apparently because this class inherits from `NamedTuple`. the error is
```
TypeError: Cannot use cached_property instance without calling __set_name__ on it.
```

I'll revert to just `@property` for now, but if there are cases where cached properties are important, we may want to consider using dataclasses instead of named tuples (and implementing indexing ourselves if needed).

Reply:
Good point! I think this method is pretty cheap as well so we can keep as is for now.